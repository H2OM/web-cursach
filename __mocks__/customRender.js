import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}