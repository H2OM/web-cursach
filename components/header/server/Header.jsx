import '../header.scss';
import Image from 'next/image';
import Link from 'next/link';
import Favmenu from '../client/Favmenuswitch';
import Favmenuswitch from '../client/Favmenuswitch';
import Searchbar from '../client/Searchbar';


export default function Header () {

    return (
        <header className='Header'>
            <div className="Header__flex container container_head">
                <Link href={"/"}>
                    <Image
                        src="/img/logo_smal.png"
                        alt="Logotype"
                        className="Header__flex__logo"
                        width={0}
                        height={0}
                        sizes='100vh'
                        priority={true}
                        quality={100}
                    />
                </Link>
                <div className="Header__flex__links">
                    <Link href="/catalog" className="smalltitle Header__flex__links__link">Каталог</Link>
                    <Link href="/news" className="smalltitle Header__flex__links__link">Новости</Link>
                    <Link href="/contacts" className="smalltitle Header__flex__links__link">Контакты</Link>
                </div>
                <div className="Header__flex__options">
                    <Searchbar/>
                    <Favmenuswitch/>
                </div>
            </div>
        </header>
    )
}