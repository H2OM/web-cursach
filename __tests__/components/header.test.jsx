import 'intersection-observer';
import customRender from '@/__mocks__/customRender';
import Header from '@/components/header/server/Header';
import { screen } from '@testing-library/react';



it('header render test', async () => {
    const {container} = customRender(<Header/>, {providerProps: {favorites: [4,3]}});
    const links = screen.getAllByRole('link');
    const images = screen.getAllByRole('img');
    
    expect(links.length).toBe(4);
    expect(images.length).toBe(3);


    expect(container).toMatchSnapshot();
});