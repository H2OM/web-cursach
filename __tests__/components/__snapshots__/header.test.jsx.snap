// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header render test 1`] = `
<div>
  <header
    class="Header"
  >
    <div
      class="Header__flex container container_head"
    >
      <a
        href="/"
      >
        <img
          alt="Logotype"
          class="Header__flex__logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="0"
          sizes="100vh"
          src="/_next/image?url=%2Fimg%2Flogo_smal.png&w=3840&q=100"
          srcset="/_next/image?url=%2Fimg%2Flogo_smal.png&w=16&q=100 16w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=32&q=100 32w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=48&q=100 48w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=64&q=100 64w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=96&q=100 96w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=128&q=100 128w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=256&q=100 256w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=384&q=100 384w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=640&q=100 640w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=750&q=100 750w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=828&q=100 828w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Flogo_smal.png&w=3840&q=100 3840w"
          style="color: transparent;"
          width="0"
        />
      </a>
      <div
        class="Header__flex__links"
      >
        <a
          class="smalltitle Header__flex__links__link"
          href="/catalog"
        >
          Каталог
        </a>
        <a
          class="smalltitle Header__flex__links__link"
          href="/news"
        >
          Новости
        </a>
        <a
          class="smalltitle Header__flex__links__link"
          href="/contacts"
        >
          Контакты
        </a>
      </div>
      <div
        class="Header__flex__options"
      >
        <div
          class="Header__flex__options__block"
        >
          <input
            class="Header__flex__options__input"
            name="searching"
            placeholder="Поиск..."
            type="text"
          />
          <img
            alt="magnifier"
            class="Header__flex__options__icon"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="30"
            src="/svg/icon_magnifier.svg"
            style="color: transparent;"
            width="30"
          />
        </div>
        <div
          class="Header__flex__options__block"
          data-count="2"
        >
          <img
            alt="star"
            class="Header__flex__options__icon Header__flex__options__icon_star"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="30"
            src="/svg/icon_star.svg"
            style="color: transparent;"
            width="30"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
